src/include/encoding.h

```diff
@@ -24,10 +24,61 @@
 // pull in the new-style encoding so that we get the denc_traits<> definition.
 #include "denc.h"
 
+
 #include "assert.h"
 
+class CephContext;
+
 using namespace ceph;
 
+namespace ceph {
+  // Occasionally an object requiring a CephContext* will need to be
+  // decoded. This object can be default constructed in every field
+  // /except/ the CephContext*. To avoid contagion (that is, to avoid
+  // having to shove a CephContext* into *every* object to preserve a
+  // uniform interface, make a specialized template function to help.
```

Comment:
We can leave it out. I just threw it in here to give people a chance to look at it and see if it was wrong-headed. The first user is PGMap inside the monitor. I'll hold it until I get that done. I just wanted to get what I had working up before master skittered away from me.